### PR TITLE
[CUDNN] Add partitioning support for conv2d and log_softmax

### PR DESCRIPTION
**Pull request by mbaret**
Further adds Relay partitioning support for cuDNN conv2d and log_softmax.

### CODE REVIEW

**Comment by mbaret** [comment]
cc @mbs-octoml @mikepapadim

File: python/tvm/relay/op/contrib/cudnn.py [hunk 1]
```diff
@@ -50,6 +51,8 @@ def partition_for_cudnn(
     tvm.IRModule
         The partitioned module.
     """
+    if params:
```
Comment (mbs-octoml) [inline python/tvm/relay/op/contrib/cudnn.py:54]: Given it's a one liner never figured out why folks want to fold that into every partition function. Cargo culting?

**Review by mbs-octoml — APPROVED** [review]
LGTM, thanks.

Not for this PR, but the direct construction of the tvm.target.cuda() here and for cublas causes some issues in the Collage branch. If there's an easy way to recover the Target already known to the te_compiler.cc machinery (via a function attribute perhaps?) that would help remove that glitch.

File: python/tvm/relay/op/contrib/cudnn.py (hunk 1 above)
Comment (masahi) [reply python/tvm/relay/op/contrib/cudnn.py:54]: In my case, I want to pattern match against a mod where batch norm is removed by constant folding + fold scale axis. Param binding is a prereq for these passes.

**Review by masahi — APPROVED** [review]

**Pull request merged**